src/main/java/uk/gov/ons/ctp/common/rest/RestUtility.java

```diff
@@ -51,24 +51,38 @@ public UriComponents createUriComponents(
     // Have to build UriComponents for query parameters separately as Expand interprets braces in
     // JSON query string
     // values as URI template variables to be replaced.
-    UriComponents uriComponents =
-        UriComponentsBuilder.newInstance()
-            .uriComponents(uriComponentsWithOutQueryParams)
-            .queryParams(queryParams)
-            .build()
-            .encode();
 
-    return uriComponents;
+    return UriComponentsBuilder.newInstance()
+        .uriComponents(uriComponentsWithOutQueryParams)
+        .queryParams(queryParams)
+        .build()
+        .encode();
   }
 
   /**
-   * Creates a Http Entity
+   * Creates a {@link HttpEntity} with basic auth header and application/json set for
+   * Content-Type and Accept
+   *
+   * @param <H> generic passed in for body content
+   * @return HttpEntity containing object as JSON
+   */
+  public <H> HttpEntity<H> createHttpEntityWithAuthHeader() {
```

Comment:
Since both `createHttpEntity...` functions include authentication headers could we just use method overloading and have this as
```
public <H> HttpEntity<H> createHttpEntity() {
  return new HttpEntity<>(getHttpHeaders());
}
```

Reply:
Yeah I was thinking this but I wanted the WithAuthHeader for context. I thought about changing the other method so they both have it but it will have a knock on impact. What d'you think?

Reply:
Yeah probably fine as it is then, definitely not worth changing the existing function since it's used so widely